sky/spot/spot_state.py

```diff
@@ -34,9 +37,36 @@
     end_at FLOAT DEFAULT NULL,
     last_recovered_at FLOAT DEFAULT -1,
     recovery_count INTEGER DEFAULT 0,
-    job_duration FLOAT DEFAULT 0)""")
+    job_duration FLOAT DEFAULT 0,
+    failure_reason TEXT,
+    new_job_id INTEGER,
+    task_id INTEGER DEFAULT 0)""")
+_CONN.commit()
 
 db_utils.add_column_to_table(_CURSOR, _CONN, 'spot', 'failure_reason', 'TEXT')
+# Create a new column for job_id, which can be duplicated for sub jobs of the
+# same job.
+# The original job_id will have no actual meaning, but only a unique identifier
+# for all jobs/sub jobs.
+db_utils.add_column_to_table(_CURSOR,
+                             _CONN,
+                             'spot',
+                             'new_job_id',
+                             'INTEGER',
+                             copy_from='job_id')
+db_utils.add_column_to_table(_CURSOR,
+                             _CONN,
+                             'spot',
+                             'task_id',
+                             'INTEGER DEFAULT 0',
+                             set_original_value=0)
+
+# `job_names` contains the mapping from job_id to the job_name.
+_CURSOR.execute("""\
+    CREATE TABLE IF NOT EXISTS job_names (
+    new_job_id INTEGER PRIMARY KEY AUTOINCREMENT,
+    name TEXT)""")
```

Comment:
Why do we need a separate table? Is this info already captured in table above?

Reply:
We need this separate table, because the `spot` table stores all tasks (falttened tasks from all the jobs). This `job_names` (renamed to `job_info`) is the table which stores the information about the jobs (one-level higher than the tasks). It currently only have the name, but we can expect that more job related information can be saved in this table in the future. If we save this job name to the `spot` table, we will have a lot duplication as every task will have one copy of the same job information.